packages/builder/plugin-swc/src/plugin.ts

```diff
@@ -98,16 +98,19 @@ export const builderPluginSwc = (
       }
 
       const isSSR = utils.target === 'node';
-      const styledComponentsOptions = applyOptionsChain(
-        getDefaultStyledComponentsConfig(isProd, isSSR),
-        config.tools.styledComponents,
-      );
-      swc.extensions.styledComponents = {
-        ...styledComponentsOptions,
-        ...(typeof swc.extensions.styledComponents === 'object'
-          ? swc.extensions?.styledComponents
-          : {}),
-      };
+
+      if (config.tools.styledComponents || swc.extensions?.styledComponents) {
```

Comment:
Can be add some e2e tests for this change?

https://github.com/web-infra-dev/modern.js/tree/main/tests/e2e/builder